enterprise/coderd/templates_test.go

```diff
@@ -873,6 +873,241 @@ func TestTemplates(t *testing.T) {
+			request.Name = "foo"
+		})
+		bar := coderdtest.CreateTemplate(t, client, user.OrganizationID, version2.ID, func(request *codersdk.CreateTemplateRequest) {
+			request.Name = "bar"
+		})
+
+		deprecationMessage := "Some deprecated message"
+		req := codersdk.UpdateTemplateMeta{
+			DeprecationMessage: &deprecationMessage,
+		}
+
+		ctx := testutil.Context(t, testutil.WaitLong)
+
+		// Deprecate foo and bar templates
+		updatedFoo, err := client.UpdateTemplateMeta(ctx, foo.ID, req)
+		require.NoError(t, err)
+		require.Equal(t, deprecationMessage, updatedFoo.DeprecationMessage)
+		updatedBar, err := client.UpdateTemplateMeta(ctx, bar.ID, req)
+		require.NoError(t, err)
+		require.Equal(t, deprecationMessage, updatedBar.DeprecationMessage)
+
+		// Should return only the deprecated templates
+		templates, err := client.Templates(ctx, codersdk.TemplateFilter{
+			SearchQuery: "deprecated:true",
+		})
+		require.NoError(t, err)
+		require.Len(t, templates, 2)
+
+		// Make sure all the deprecated templates are returned
+		expectedTemplates := map[uuid.UUID]codersdk.Template{
+			updatedFoo.ID: updatedFoo,
+			updatedBar.ID: updatedBar,
+		}
+		actualTemplates := map[uuid.UUID]codersdk.Template{}
+		for _, template := range templates {
+			actualTemplates[template.ID] = template
+		}
+
+		require.Equal(t, len(expectedTemplates), len(actualTemplates))
+		for id, expectedTemplate := range expectedTemplates {
+			actualTemplate, ok := actualTemplates[id]
+			require.True(t, ok)
+			require.Equal(t, expectedTemplate.ID, actualTemplate.ID)
+			require.Equal(t, true, actualTemplate.Deprecated)
+			require.Equal(t, expectedTemplate.DeprecationMessage, actualTemplate.DeprecationMessage)
+		}
+	})
+
+	// Should return only non-deprecated templates when filtering by deprecated=false
+	t.Run("ListMultiple deprecated=false", func(t *testing.T) {
+		t.Parallel()
+
+		owner, user := coderdenttest.New(t, &coderdenttest.Options{
+			LicenseOptions: &coderdenttest.LicenseOptions{
+				Features: license.Features{
+					codersdk.FeatureAccessControl: 1,
+				},
+			},
+		})
+		client, _ := coderdtest.CreateAnotherUser(t, owner, user.OrganizationID, rbac.RoleTemplateAdmin())
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+		version2 := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+		foo := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID, func(request *codersdk.CreateTemplateRequest) {
+			request.Name = "foo"
+		})
+		bar := coderdtest.CreateTemplate(t, client, user.OrganizationID, version2.ID, func(request *codersdk.CreateTemplateRequest) {
+			request.Name = "bar"
+		})
+
+		ctx := testutil.Context(t, testutil.WaitLong)
+
+		// Should return only the deprecated templates
+		templates, err := client.Templates(ctx, codersdk.TemplateFilter{
+			SearchQuery: "deprecated:false",
+		})
```

Comment:
Potential copy-pasta comment here. This should only return **non-deprecated** templates.

Reply:
Great catch! 👀  Updating...
```suggestion
		// Should return only the non-deprecated templates
		templates, err := client.Templates(ctx, codersdk.TemplateFilter{
			SearchQuery: "deprecated:false",
		})
```